scripts/operatorhub-deploy.sh

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Kohl's Department Stores, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+# Colors for better user experience
+WHITE='\033[1;37m'
+NC='\033[0m' # No color
+
+# This script helps in deployment of new version of Eunomia to OperatorHub
+usage() {
+    cat <<EOT
+operatorhub-deploy.sh VERSION
+
+VERSION in format x.y.z, e.g. 0.1.2
+
+Create a pull request in https://github.com/operator-framework/community-operators
+in order to add Eunomia in VERSION to OperatorHub.
+EOT
+}
+
+[[ -z "${1:-}" ]] && {
+    echo "Missing argument"
+    usage
+    exit 1
+}
+[[ $1 =~ ^[0-9]{1,2}\.[0-9]{1,2}\.[0-9]{1,2}$ ]] || {
+    echo "VERSION in wrong format"
+    usage
+    exit 1
+}
+new_version="$1"
+eunomia_root=$(
+    cd $(dirname "$0")/..
+    pwd
+)
+
+# Satisfy "operator-sdk olm-catalog gen-csv"
+# Create operator.yaml and role.yaml
+echo -e "${WHITE}Generating files to satisfy operator-sdk"
+echo -e "\n- Generating operator.yaml and role.yaml${NC}"
+helm template "${eunomia_root}/deploy/helm/eunomia-operator/" \
+    --set eunomia.operator.deployment.operatorHub=true \
+    --set eunomia.operator.image.tag="v${new_version}" \
+    --output-dir "${eunomia_root}/deploy"
+# The intermediate directories and files must be created, no workaround for this
+cat "${eunomia_root}/deploy/eunomia-operator/templates/deployment.yaml" \
+    >"${eunomia_root}/deploy/operator.yaml"
+cat "${eunomia_root}/deploy/eunomia-operator/templates/clusterrole-operators.yaml" \
+    "${eunomia_root}/deploy/eunomia-operator/templates/role-operator.yaml" \
+    >"${eunomia_root}/deploy/role.yaml"
+# Delete artifacts
+rm -rf "${eunomia_root}/deploy/eunomia-operator/"
+
+# Get previous versions from OperatorHub
+echo -e "\n${WHITE}- Downloading previous releases of Eunomia from OperatorHub${NC}"
+mkdir -p "${eunomia_root}/deploy/olm-catalog"
+# Why svn must be used, not git-archive:
+# https://stackoverflow.com/a/18194523
+# https://twitter.com/GitHubHelp/status/322818593748303873
+(
+    cd "${eunomia_root}/deploy/olm-catalog" &&
+        svn checkout https://github.com/operator-framework/community-operators/trunk/upstream-community-operators/eunomia &&
```

Comment:
[optional] IIUC, on l.125 we're still forking whole community-operators directory anyway; couldn't we just do it earlier here, and just copy a subdirectory from it, and drop the svn dependency?